js/compiler.ts

```diff
@@ -196,8 +197,16 @@ export class DenoCompiler
         "Cannot run module without factory."
       );
       assert(moduleMetaData.hasRun === false, "Module has already been run.");
+      // while TypeScript always transpiles to modify the export object, to work
+      // with all types of AMD modules, we have to detect a return value from
+      // the factory
       // asserts not tracked by TypeScripts, so using not null operator
-      moduleMetaData.factory!(...this._getFactoryArguments(moduleMetaData));
+      const exports = moduleMetaData.factory!(
+        ...this._getFactoryArguments(moduleMetaData)
+      );
+      if (exports != null) {
```

Comment:
Move above comment here. Mention that this is for JSON support (and potentially other things in the future)/